src/others/sysmon/lv_sysmon.c

```diff
@@ -84,19 +70,16 @@ void _lv_sysmon_builtin_init(void)
 
 void _lv_sysmon_builtin_deinit(void)
 {
-#if _USE_PERF_MONITOR
-    lv_timer_delete(sysmon_perf.timer);
-#endif
-
-#if _USE_MEM_MONITOR
+#if LV_USE_MEM_MONITOR
     lv_timer_delete(sysmon_mem.timer);
 #endif
 }
 
-lv_obj_t * lv_sysmon_create(lv_obj_t * parent)
+lv_obj_t * lv_sysmon_create(lv_display_t * disp)
 {
     LV_LOG_INFO("begin");
-    lv_obj_t * label = lv_label_create(parent);
+    if(disp == NULL) disp = lv_display_get_default();
```

Comment:
There needs to be a second check to see if the display is NULL. If it is then an error should be made to the log and the function should exit.

```c
if(disp == NULL) {
    disp = lv_display_get_default();
    if(disp == NULL) {
        // log error
        return NULL;
    }
}
```

I thought that display_get_default can return a NULL pointer if there is no display initialized. If I am right about that then this will need to be done.